src/pip/_internal/utils/logging.py

```diff
@@ -51,8 +55,62 @@
     colorama = _colorama
 
 
+@python_2_unicode_compatible
+class LogMessage:
+    """Lazily interpolated log message formatted using str.format."""
+    def __init__(self, message, args):
+        # type: (str, Tuple[Any, ...]) -> None
+        self.message, self.args = message, args
+
+    def __str__(self):
+        # type: () -> str
+        return self.message.format(*self.args)
+
+
+class FormatLogger(logging.Logger):
+    """Logger adapter that uses str.format for messages formatting.
+
+    Skip formatting if no varadic argument is passed to the methods.
+    """
+    def __init__(self, name):
+        # type: (str) -> None
+        super(FormatLogger, self).__init__(name)
+
+    def log(self, level, msg, *args, **kwargs):
+        # type: (int, str, *Any, **Any) -> None
+        message = LogMessage(msg, args) if args else msg
+        super(FormatLogger, self).log(level, message, **kwargs)
+
+    def debug(self, msg, *args, **kwargs):
+        # type: (str, *Any, **Any) -> None
+        self.log(logging.DEBUG, msg, *args, **kwargs)
+
+    def info(self, msg, *args, **kwargs):
+        # type: (str, *Any, **Any) -> None
+        self.log(logging.INFO, msg, *args, **kwargs)
+
+    def warning(self, msg, *args, **kwargs):
+        # type: (str, *Any, **Any) -> None
+        self.log(logging.WARNING, msg, *args, **kwargs)
+
+    def error(self, msg, *args, **kwargs):
+        # type: (str, *Any, **Any) -> None
+        self.log(logging.ERROR, msg, *args, **kwargs)
+
+    def exception(self, msg, *args, **kwargs):
+        # type: (str, *Any, **Any) -> None
+        kwargs['exc_info'] = True
+        self.log(logging.ERROR, msg, *args, **kwargs)
+
+    def critical(self, msg, *args, **kwargs):
+        # type: (str, *Any, **Any) -> None
+        self.log(logging.CRITICAL, msg, *args, **kwargs)
+
+
 _log_state = threading.local()
+logging.setLoggerClass(FormatLogger)
```

Comment:
I don't think we can make this change unilaterally, because some of our dependencies may also be using `logging` with the normal format. At the very least it needs to be scoped to `pip` loggers (other than ones under `pip._vendor`).